langchain/src/document_loaders/web/wikipedia.ts

```diff
@@ -0,0 +1,131 @@
+import wiki, { Page } from "wikipedia";
+import { Document } from "../../document.js";
+import { BaseDocumentLoader } from "../base.js";
+
+const DISCARDCATEGORIES = [
+  "See also",
+  "References",
+  "External links",
+  "Further reading",
+  "Footnotes",
+  "Bibliography",
+  "Sources",
+  "Citations",
+  "Literature",
+  "Footnotes",
+  "Notes and references",
+  "Photo gallery",
+  "Works cited",
+  "Photos",
+  "Gallery",
+  "Notes",
+  "References and sources",
+  "References and notes",
+];
+
+export class WikipediaLoader extends BaseDocumentLoader {
+  discardCategories: string[];
+
+  constructor(public webPath: string, discardCategories?: string[]) {
```

Comment:
To match the feature parity of `wikipedia` loader in Langchain, it would make sense to add an option for locale via `wiki.setLang(lang)`